### PR TITLE
CLC-6199 Fix reactivation of bCourses account when SIS ID is also changed

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-6199

I expect this will migrate to a module method at some point to be used from other classes, but didn't want to delay the fix.

### CODE REVIEW

File: app/models/canvas_csv/maintain_users.rb [hunk 1]
```diff
@@ -28,27 +28,34 @@ def self.change_sis_user_id(canvas_user_id, new_sis_user_id)
...
       end
       false
     end
 
+    def self.parse_login_id(login_id)
+      if (matched = /^(inactive-)?([0-9]+)$/.match login_id)
+        inactive_account = matched[1]
+        ldap_uid = matched[2].to_i
+      else
+      end
```
Comment (johncrossman) [inline app/models/canvas_csv/maintain_users.rb]: @raydavis Empty else block?

Comment (raydavis) [inline app/models/canvas_csv/maintain_users.rb]: Yeah. So what's your point?

... amendment coming, thanks!

File: app/models/canvas_csv/maintain_users.rb [hunk 2]
```diff
@@ -131,22 +137,21 @@ def handle_email_deletions(canvas_user_ids)
...
       old_account_data = existing_account.to_hash
-      login_id = old_account_data['login_id']
-      if (inactive_account = /^inactive-([0-9]+)$/.match login_id)
-        login_id = inactive_account[1]
-      end
-      if (ldap_uid = Integer(login_id, 10) rescue nil)
-        @known_uids << login_id
+      parsed_login_id = self.class.parse_login_id old_account_data['login_id']
+      ldap_uid = parsed_login_id[:ldap_uid]
+      inactive_account = parsed_login_id[:inactive_account]
```
Comment (johncrossman) [inline app/models/canvas_csv/maintain_users.rb]: @raydavis Not a big deal: It seems we only care about `inactive_account` value if ldap_uid is not nil. Putting the expression above within the `if ldap_uid` block would make that clear. Not a merge blocker.

**Comment by johncrossman** [comment]
Looks good to me, @raydavis

File: app/models/canvas_csv/maintain_users.rb (hunk 2 above)
Comment (raydavis) [inline app/models/canvas_csv/maintain_users.rb]: I vacillated on that myself, @johncrossman , and decided to keep the parsed_login_id hash out of the inner block logic instead.

**Pull request merged**